### ISSUE TITLE
Support Greasemonkey user-scripts

### ISSUE THREAD

**Issue by aartaka** [atlas-engineer/nyxt#2021]
Greasemonkey scripts are a way to alter page behavior with a granular control of where, when, and how the script is injected. There are dozens of scripts around the web for lots of purposes, so having support for those can help our users to customize the Web by simply loading necessary scripts into Nyxt. A good way to start customization :)

We have `ffi-buffer-add-user-script` and WebKitGTK allows quite some customization in it. What we need to enable Greasemonkey-like scripts is 
- [Greasemonkey metadata](https://wiki.greasespot.net/Metadata_Block) parsing and 
- a convenient interface for script addition.

**Comment by guojing0** [comment]
I am not sure about current status of WebExtensions (WE). If WE is supported, then could not we just add Greasemonkey along with user scripts?

**Comment by aartaka** [comment]
Yes, that makes perfect sense. However, having alternative interfaces and ways to do things that are more renderer-agnostic than WebExtensions (those lock us to WebKitGTK at the moment) would be good, especially if we ever resurrect Qt port.

**Comment by aartaka** [comment]
Done with #2139.
